state_transfer/rclone_transfer_server.go

```diff
@@ -0,0 +1,150 @@
+package state_transfer
+
+import (
+	"context"
+	random "math/rand"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	rcloneServerConf = `[mnt]
+type = local
+`
+)
+
+func (r *RcloneTransfer) createTransferServerResources(c client.Client) error {
+	var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	random.Seed(time.Now().UnixNano())
+	password := make([]byte, 24)
+	for i := range password {
+		password[i] = letters[random.Intn(len(letters))]
+	}
+
+	r.SetPassword(string(password))
+	r.SetPort(rclonePort)
+	r.SetUsername(rcloneUser)
+
+	err := createRcloneServerConfig(c, r)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createRcloneServerConfig(c client.Client, r *RcloneTransfer) error {
+	rcloneConfigMap := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.PVC().Namespace,
+			Name:      rcloneConfigPrefix + r.PVC().Name,
```

Comment:
you probably want to be using something more narrow-scoped for not running into shared config with same named PVC accross two namespaces.

DVM uses `md5hash(rcloneConfigPrefix + r.PVC().Name + r.PVC().Namespace)`, we got bugs in DVM where PVC name was the same across two namespaces and the config was shared and failed DVM

Reply:
I don't fully understand, but sure, if this produces something unique, it works for me. That's all I was after.